### PR TITLE
Custodian donator item

### PR DESCRIPTION
**Pull request by GDRdev**
Adds a set of custodian armor for my donator perk

(havent tested it)

(Armor)
CustodianFront

CustodianBackchest

Custodianchestleft

Custodianchestright

(Helmet)

CustodianHatrightt

CustodianHatLeftt

Custodianhatback

CustodianBack

# Wiki Documentation = nothing

This is basically a copy and paste from another donator pull request soo hopefully it will work (I dont know WHY code\modules\antagonists\sentient_creature.dm got moved and I had no idea how to fix it or when that happened. Github is actually worst then dream maker to code I swear.

# Changelog

:cl:
rscadd: Custodian donor items
/:cl:

### CODE REVIEW

**Review by Moltijoe — CHANGES_REQUESTED** [review]
there's a number of locations you're using capitals, please use lowercase.
item name, typepaths, as well as all sprite names should be in lower case.
also, if you're adding multiple items, instead create a box that contains both items that is given via donator_gear

I have no clue about the validity of giving "fake armour" as donator gear, but that's not my jurisdiction

**Comment by MajManatee** [comment]
Refrence #17777 on ways to create and use the box mentioned above.

**Comment by GDRdev** [comment]
There we are updated! I think I got rid of the things with unneeded caps and I THINK I put the set into a box

**Comment by ynot01** [comment]
Merge conflicts

**Comment by GDRdev** [comment]
> Merge conflicts

uuhhhh how do I fix that?

**Comment by GDRdev** [comment]
> Merge conflicts

okay NOW I think its fixed but you will have to check if I did or not cuz I have no idea

File: yogstation/code/modules/clothing/donator.dm [hunk 1]
```diff
@@ -189,14 +189,26 @@
...
+/obj/item/storage/box/somecustodian
+	name = "Golden Box"
+	desc = "Holds a set of one of the most powerful suits of armor known to mankind.(99 cents rip off)"
+
+/obj/item/storage/box/somecustodian/PopulateContents()
+	. = ..()
+	new /obj/item/clothing/head/custodianhelm(src)
+	new /obj/item/clothing/suit/custodianchest(src)
+
+/obj/item/bedsheets/pridebedsheet/cark
```
Comment (ynot01) [inline yogstation/code/modules/clothing/donator.dm]: these changes are not a part of your pr

File: yogstation/code/modules/clothing/donator.dm [hunk 2]
```diff
@@ -189,14 +189,26 @@
...
 /obj/item/bedsheet/pride
 	name = "Pride Bedsheet"
 	desc = "A very prideful queen size bedsheet. How you are carrying all that weight is beyond me though."
 	mob_overlay_icon = 'icons/mob/clothing/neck/sheets.dmi'
+	lefthand_file = 'icons/mob/inhands/misc/bedsheet_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/bedsheet_righthand.dmi'
```
Comment (ynot01) [inline yogstation/code/modules/clothing/donator.dm]: ditto

File: yogstation/code/modules/clothing/donator.dm [hunk 3]
```diff
@@ -189,14 +189,26 @@
...
 
-/obj/item/storage/box/boxta/PopulateContents() 
+/obj/item/storage/box/boxta/PopulateContents()
 	. = ..()
 	new /obj/item/clothing/neck/falcon(src)
 	new /obj/item/clothing/head/fedora/gtrim_fedora(src)
 
+/obj/item/storage/box/somecustodian
+	name = "Golden Box"
+	desc = "Holds a set of one of the most powerful suits of armor known to mankind.(99 cents rip off)"
```
Comment (ynot01) [inline yogstation/code/modules/clothing/donator.dm]: spacing and i don't think the parenthesis will look good in a description at regardless

File: yogstation.dme [hunk 4]
```diff
@@ -1466,6 +1466,7 @@
 #include "code\modules\admin\view_variables\topic_basic.dm"
 #include "code\modules\admin\view_variables\topic_list.dm"
 #include "code\modules\admin\view_variables\view_variables.dm"
+#include "code\modules\antagonists\sentient_creature.dm"
```
Comment (ynot01) [inline yogstation.dme]: ?

**Review by ynot01 — CHANGES_REQUESTED** [review]

File: yogstation.dme (hunk 4 above)
Comment (GDRdev) [reply yogstation.dme]: this I have no idea why that happen

File: yogstation/code/modules/clothing/donator.dm (hunk 3 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: fair

File: yogstation/code/modules/clothing/donator.dm (hunk 2 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: I dont know where it was supposed to go

File: yogstation/code/modules/clothing/donator.dm (hunk 1 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: top part is but I was just trying to fix my mistake

File: yogstation/code/modules/clothing/donator.dm (hunk 2 above)
Comment (ynot01) [reply yogstation/code/modules/clothing/donator.dm]: why are you touching the pride bedsheet at all

File: yogstation/code/modules/clothing/donator.dm (hunk 1 above)
Comment (ynot01) [reply yogstation/code/modules/clothing/donator.dm]: remove the `/obj/item/bedsheets/pridebedsheet/cark` line

File: yogstation.dme (hunk 4 above)
Comment (ynot01) [reply yogstation.dme]: undo this change

File: yogstation/code/modules/clothing/donator.dm (hunk 2 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: it was a "conflict" and somehow some of my code was broken inside it? soo I guess removed the code that was mine

File: yogstation.dme (hunk 4 above)
Comment (GDRdev) [reply yogstation.dme]: I dont know how...

File: yogstation/code/modules/clothing/donator.dm (hunk 2 above)
Comment (ynot01) [reply yogstation/code/modules/clothing/donator.dm]: remove the changed lines

File: yogstation.dme (hunk 4 above)
Comment (ynot01) [reply yogstation.dme]: move it back to where it was

File: yogstation/code/modules/clothing/donator.dm (hunk 2 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: just totally remove them? alright

File: yogstation.dme (hunk 4 above)
Comment (GDRdev) [reply yogstation.dme]: I dont know where is the issue

**Comment by GDRdev** [comment]
there we are I think I have fixed everything that ynot wanted me to fix

File: yogstation.dme (hunk 4 above)
Comment (ThatLing) [reply yogstation.dme]: @ynot01 This happens when you open the project in DM, the include order got fucked at some point

File: code/modules/clothing/head/misc_special.dm [hunk 5]
```diff
@@ -372,6 +372,12 @@
 	item_state = "rflower"
 	w_class = WEIGHT_CLASS_TINY
 
+/obj/item/clothing/head/custodianhelm
+	name = "Custodian Helmet"
```
Comment (ynot01) [inline code/modules/clothing/head/misc_special.dm]: lowercase all of this; if Custodian is a proper noun then prefix it with \improper ie "\improper Gygax chassis"

File: code/modules/clothing/head/misc_special.dm [hunk 6]
```diff
@@ -372,6 +372,12 @@
...
+	desc = "Gives you the willpower to defend your plasmaman Captain till the end of time. If this was real that is..."
+	icon_state = "CustodianHat"
```
Comment (ynot01) [inline code/modules/clothing/head/misc_special.dm]: lowercase icon name

File: code/modules/clothing/suits/miscellaneous.dm [hunk 7]
```diff
@@ -366,6 +366,12 @@
  * Misc
  */
 
+/obj/item/clothing/suit/custodianchest
+	name = "Custodian Armor"
```
Comment (ynot01) [inline code/modules/clothing/suits/miscellaneous.dm]: same as above

File: code/modules/clothing/suits/miscellaneous.dm [hunk 8]
```diff
@@ -366,6 +366,12 @@
...
+	desc = "Strong enough to handle the toughest of blows if it was made with the real deal instead of some trash material like gold"
```
Comment (ynot01) [inline code/modules/clothing/suits/miscellaneous.dm]: description needs to be punctuated

File: code/modules/clothing/suits/miscellaneous.dm [hunk 9]
```diff
@@ -366,6 +366,12 @@
...
+	icon_state = "CustodianBody"
```
Comment (ynot01) [inline code/modules/clothing/suits/miscellaneous.dm]: lowercase icon name

File: yogstation/code/modules/clothing/donator.dm [hunk 10]
```diff
@@ -189,11 +189,20 @@
...
 
-/obj/item/storage/box/boxta/PopulateContents() 
+/obj/item/storage/box/boxta/PopulateContents()
 	. = ..()
 	new /obj/item/clothing/neck/falcon(src)
 	new /obj/item/clothing/head/fedora/gtrim_fedora(src)
 
+/obj/item/storage/box/somecustodian
+	name = "Golden Box"
+	desc = "Holds a set of one of the most powerful suits of armor known to mankind. Of course its the 99 cents rip off verison"
```
Comment (ynot01) [inline yogstation/code/modules/clothing/donator.dm]:
```suggestion
	desc = "Holds a set of one of the most powerful suits of armor known to man. A '99 cents' price tag hangs off the corner."
```

**Review by ynot01 — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/clothing/donator.dm [hunk 11]
```diff
@@ -189,11 +189,20 @@
...
 	desc = "A criminals' belongings, returned upon their release. Just stop killing people."
 
-/obj/item/storage/box/boxta/PopulateContents() 
+/obj/item/storage/box/boxta/PopulateContents()
 	. = ..()
 	new /obj/item/clothing/neck/falcon(src)
 	new /obj/item/clothing/head/fedora/gtrim_fedora(src)
 
+/obj/item/storage/box/somecustodian
+	name = "Golden Box"
```
Comment (ynot01) [inline yogstation/code/modules/clothing/donator.dm]: since this is apparently a cheap toy this should be improper

**Comment by ReddicusDragon** [comment]
I am of the opinion that donors should only get 1 custom item.

File: yogstation/code/modules/clothing/donator.dm (hunk 11 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: Piss

**Comment by GDRdev** [comment]
> I am of the opinion that donors should only get 1 custom item.

Fair but others have gotten more then one item in the past soo meh. Atleast its not giving me perks or whatever.

**Comment by GDRdev** [comment]
I do think everything has been fixed that was needed to be fixed and its PROBABLY fine to merge

File: code/modules/clothing/suits/miscellaneous.dm [hunk 12]
```diff
@@ -366,6 +366,12 @@
  * Misc
  */
 
+/obj/item/clothing/suit/custodianchest
+	name = "custodian armor"
+	desc = "Strong enough to handle the toughest of blows! If it was made with the real deal instead of some trash material like gold"
```
Comment (ynot01) [inline code/modules/clothing/suits/miscellaneous.dm]: Description needs to end in punctuation

File: yogstation/code/modules/clothing/donator.dm [hunk 13]
```diff
@@ -189,11 +189,20 @@
...
 	desc = "A criminals' belongings, returned upon their release. Just stop killing people."
 
-/obj/item/storage/box/boxta/PopulateContents() 
+/obj/item/storage/box/boxta/PopulateContents()
 	. = ..()
 	new /obj/item/clothing/neck/falcon(src)
 	new /obj/item/clothing/head/fedora/gtrim_fedora(src)
 
+/obj/item/storage/box/somecustodian
+	name = "piss colored box"
```
Comment (ynot01) [inline yogstation/code/modules/clothing/donator.dm]: Sus
Also would this not be "piss-colored box"

**Review by ynot01 — CHANGES_REQUESTED** [review]
You changed the icon names in code but haven't changed the icon file to reflect
Also the helmet sprite flickers

File: code/modules/clothing/suits/miscellaneous.dm (hunk 12 above)
Comment (GDRdev) [reply code/modules/clothing/suits/miscellaneous.dm]: oh alright

File: yogstation/code/modules/clothing/donator.dm (hunk 13 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: > You changed the icon names in code but haven't changed the icon file to reflect Also the helmet sprite flickers

to the icon names yes I actually did rename them

Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: > Sus Also would this not be "piss-colored box"

Its probably fine as is but I will change it

**Comment by GDRdev** [comment]
> You changed the icon names in code but haven't changed the icon file to reflect Also the helmet sprite flickers

fixed the flicker

File: yogstation/code/modules/clothing/donator.dm (hunk 13 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: refresh?

>

**Comment by GDRdev** [comment]
Now everything should be fixed

File: yogstation/code/modules/clothing/donator.dm (hunk 13 above)
Comment (GDRdev) [reply yogstation/code/modules/clothing/donator.dm]: image

**Comment by GDRdev** [comment]
every fucking time I try to do something
image

**Comment by MajManatee** [comment]
Will likely have to wait until the current donator item stuff is resolved.

**Comment by MajManatee** [comment]
Also, please test your code.

**Comment by GDRdev** [comment]
> Also, please test your code.

no idea how

**Comment by ynot01** [comment]
https://wiki.yogstation.net/wiki/Downloading_the_source_code#Hosting_a_server

**Review by ynot01 — CHANGES_REQUESTED** [review]
Why do you have duplicates

Also merge conflicts

**Comment by JamieD1** [comment]
On hold due to ongoing discussions regarding donor items

**Review by cuackles — CHANGES_REQUESTED** [review]
ok so sprite review sprite review, first of all i would like for you to post ingame photos of the sprite, to see how well it goes with everything else, apart from that some things about the sprite, 
image
i would like it for the shoulder pads to have a consistent shading, making the side sprites have the same as the front and back one, apart from that the ass could use a bit more defining so it aint looking like a flat.
for the helmet
image
make sure the ponytail casts a shadow on the back of the head, apart from that maybe add a bit of depth to the front face so it feels a bit less flat but apart from that its good.
last thing is, i haven't checked if its it or not, but you should make it so this is a suit that can deploy its helmet instead of making them 2 separate items for the sake of simplicity if it isn't already that

**Comment by MajManatee** [comment]
@GDRdev DMI conflicts. 2 Weeks no pushes / changes made that were requested. Please re-open when work continutes

**Comment by Hopekz** [comment]
> @GDRdev DMI conflicts. 2 Weeks no pushes / changes made that were requested. Please re-open when work continutes

Jamie wrote "On hold due to ongoing discussions regarding donor items" 2 posts above

**Pull request closed without merging**